waku/v2/protocol/filter/filter_health_check.go

```diff
@@ -24,7 +24,7 @@ func (wf *WakuFilterLightNode) PingPeer(peer peer.ID) {
 	ctxWithTimeout, cancel := context.WithTimeout(wf.CommonService.Context(), PingTimeout)
 	defer cancel()
 	err := wf.Ping(ctxWithTimeout, peer)
-	if err != nil {
+	if err != nil && wf.onlineChecker.IsOnline() {
```

Comment:
I wonder if we should instead do
```
if wf.onlineChecker.IsOnline() {
   err := wf.Ping(ctxWithTimeout, peer)
   if err != nil {
              ...............
   }
}
```
to not even attempt to ping if the node is offline

Reply:
that is already handled in https://github.com/waku-org/go-waku/blob/e0a9ae0447cecfe3e301f33a1c6b8509a2bb59b0/waku/v2/protocol/filter/filter_health_check.go#L52-L55.

While going through the logs i had noticed that many times ping goes through and then node goes offline and failures cause issue, hence i added this additional check while checking for ping result. It is not completely fool-proof but it does take care of additional scenarios.

I did not want to complicate the code further by tying up ping context to online status, hence left it at this.